core/test/unit/services/url/utils_spec.js

```diff
@@ -83,6 +83,13 @@ describe('Url', function () {
         });
     });
 
+    describe('deduplicateSubDir', function () {
```

Comment:
> urlService.utils.deduplicateSubDir('http://myblog.com/blog/blog/').should.eql('http://myblog.com/blog/blog/');

This doesn't work. I think you need to modify the `deduplicateSubDir` fn and not the `getSubdir` fn. `getSubdir` just returns the subdir if you have configured one.

Reply:
Sorry but I don't get it. What doesn't work here? `deduplicateSubDir` passed its test and it didn't remove `blog` subdir from original link, is it expected behavior?

Reply:
> urlService.utils.deduplicateSubDir('http://myblog.com/blog/blog/').should.eql('http://myblog.com/blog/blog/');

With trailing slashes.

Reply:
Got it, didn't notice the trailing slash. It failed with trailing slash but I don't know what's purpose of this function. Why 2 subdirs with same name are not allowed, can you explain?

Reply:
This issue is about defining a post with the slug "blog" and a subdirectory `/blog/`.

`http://localhost:8888/blog/blog/` is the url of the post in this case. But it doesn't work, because `deduplicateSubDirs` fn interprets this url as "two subdirectories".

Reply:
I see this case (post slug issue). My question was about purpose of function `deduplicateSubDirs`, why does it worry about 2 subdirs? and in what case a user can configure 2 subdirs for a Ghost instance e.g. http://myblog.com/blog/blog? and even he does it, what's wrong?

Reply:
This was probably added in the past to ensure:

- if you would like to join two urls and you accidentally pass both urls with subdirectory

Originally added here https://github.com/TryGhost/Ghost/commit/2700bfa4ccbe151b098bd0e694a81db9f4d12be9.

Reply:
@tiendq ping :)